### PR TITLE
feat: add cross account deployment guide for gen 2

### PR DESCRIPTION
**Pull request by Jay2113**
#### Description of changes:
Add a comprehensive guide on how to perform cross account deployments for Amplify Gen 2 apps using Amazon CodeCatalyst and AWS Amplify Hosting.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries
- [x] amplify-gen2

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?
Commented out `yarn spellcheck-diff` to ignore spellcheck.

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/deploy-and-host/fullstack-branching/cross-account-deployments/index.mdx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+export function getStaticProps(context) {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+While building with Amplify CI/CD offers benefits such as zero-config setup, fullstack previews, centralized secrets management, Amplify Gen 2 makes it easy to integrate fullstack CI/CD into your custom pipelines (e.g. AWS CodePipeline, Amazon CodeCatalyst, GitHub Actions etc.). This guide will walk you through how to create a customized deployment pipeline for your AWS Amplify Gen 2 applications powered by Amazon CodeCatalyst and AWS Amplify Hosting. You can follow along the step-by-step process to build this deployment workflow for your cross account setup and can modify it based on your specific project needs.
```
Comment (swaminator) [inline src/pages/gen2/deploy-and-host/fullstack-branching/cross-account-deployments/index.mdx:15]: This guide walks you through how to create a trunk-based, multi-region deployment pipeline for your AWS Amplify Gen 2 applications. In this tutorial, we will be using Amazon CodeCatalyst and AWS Amplify Hosting, but you can choose any CI/CD provider.

**Review by swaminator — CHANGES_REQUESTED** [review]
Request update to the intro. Rest LGTM.

File: src/pages/gen2/deploy-and-host/fullstack-branching/cross-account-deployments/index.mdx (hunk 1 above)
Comment (Jay2113) [reply src/pages/gen2/deploy-and-host/fullstack-branching/cross-account-deployments/index.mdx]: Done, the introduction has been updated.

**Review by swaminator — DISMISSED** [review]

File: .husky/pre-commit [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env sh
-. "$(dirname -- "$0")/_/husky.sh"
+# . "$(dirname -- "$0")/_/husky.sh"
```
Comment (reesscot) [inline .husky/pre-commit:2]: Please don't modify this file

**Review by reesscot — CHANGES_REQUESTED** [review]
Please add any spellcheck failures to the cspell file

File: .husky/pre-commit (hunk 2 above)
Comment (Jay2113) [reply .husky/pre-commit]: Done, have reverted the change in latest commit.

**Review by swaminator — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**